store/src/java/com/zimbra/cs/service/mail/ItemAction.java

```diff
@@ -200,6 +211,11 @@ protected ItemActionResult handleCommon(Map<String, Object> context, Element req
                 localResults = ItemActionHelper.DUMPSTER_DELETE(octxt, mbox, responseProto, local, type, tcon).getResult();
             } else if (opStr.equals(MailConstants.OP_TRASH)) {
                 localResults = handleTrashOperation(octxt, request, mbox, responseProto, local, type, tcon);
+                try {
+                    handleTrashOperationForSharedFile(octxt, mbox, local, type, context);
+                } catch (ServiceException | MessagingException e) {
+                    throw ServiceException.FAILURE("Error performing handleTrashOperationForSharedFile", e);
```

Comment:
@gabhinavTI since the delete operation is done on #213 and now if we get exception during revoke for shared file, will still have shared record on sharee's end, can you check this scenario and what happens in this case to actual file on sharer's end

Reply:
@dasiyogesh
Since these are 2 different operations and not dependent on each other
so When operation on 213 run ```handleTrashOperation()```  runs, it would first delete the file from the owner's end
and when the second operation runs ```handleTrashOperationForSharedFile()``` then it would delete the file at the receiver end

Now as per your scenario, the Actual file will be deleted from the owner end and the file at the grantee end is visible but would throw an exception that the file can not be previewed / not found exception